lib/core/diagnostics.js

```diff
@@ -114,6 +120,76 @@ if (undiciDebugLog.enabled || fetchDebuglog.enabled) {
   isClientSet = true
 }
 
+// Track fetch requests
+if (fetchDebuglog.enabled) {
+  const debuglog = fetchDebuglog.enabled
+
+  diagnosticsChannel.channel('undici:fetch:start').subscribe(evt => {
+    const {
+      input
+    } = evt
+    debuglog(
+      'fetch request has started request to %s',
+      input
+    )
+  })
+
+  diagnosticsChannel.channel('undici:fetch:end').subscribe(evt => {
+    const {
+      input,
+      error
+    } = evt
+    debuglog(
+      'received response from %s, error %s',
+      input,
+      error?.message
+    )
+  })
```

Comment:
I have the feeling, that the expectation here is, that input is always a stream. If it is an URL object, or worse a huuge Request.Object, do we want to have it completely serialized into the debugLog?